modules/standard/Spawn.chpl

```diff
@@ -668,10 +748,53 @@ module Spawn {
   proc subprocess.communicate() {
     var err:syserr = ENOERR;
 
+    _halt_on_launch_error();
+
     this.communicate(error=err);
-    if err then halt("Error in subprocess.communicate: " + errorToString(err));
+    if err then ioerror(err, "in subprocess.communicate");
   }
 
+  /*
+    Close any open channels and pipes for interacting with a subprocess.  This
+    function does not wait for the subprocess to complete.  Note that it is
+    generally not necessary to call this function since these channels will be
+    closed when the subprocess record goes out of scope.
```

Comment:
That's different than our strategy for other types, correct?  What's the precedent for the change?